### PR TITLE
Changes collector URL property to URI

### LINKED ISSUE #1367: Remove path from metrics collector URL properties
Currently we are setting the property `spring.cloud.dataflow.metrics.collector.url` to have the full server path including `/collector/metrics/streams` 
We should just use the server and port and leave the resolution of the right path to the client part. 
We can break in a two fold change, first we set it on the client, but later we should upgrade to Traverson to find the proper link and follow it.

### PR DESCRIPTION
**Pull request by viniciusccarvalho**
Fixes #1367

### CODE REVIEW

**Comment by jvalkeal** [comment]
property is `URL`, not `URI`. `URI` can also be `URN` which we never use. And besides, boot properties are usually named as `url`. Even in `RestTemplate` methods parameters are named `String url`.

File: spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc [hunk 1]
```diff
@@ -801,11 +801,21 @@ Data Flow Metrics Collector, and the Data Flow server is shown below.
...
-Metrics Collector for Kafka 0.9, Kafka 0.10, and RabbitMQ.  You can find
-more information on building and running the Metrics Collector on its
+
+Metrics Collector for all of the supported binders.  You can find
+more information on building and running the Metrics Collector on its 
+
 https://github.com/spring-cloud/spring-cloud-dataflow-metrics-collector[project
 page].
 
+The dataflow server now accepts an optional property `spring.cloud.dataflow.metrics.collector.uri`, this property should point to the URI of your deployed
```
Comment (markpollack) [inline spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc]: introduce `\n` at ~80 cols in future.

**Comment by viniciusccarvalho** [comment]
@jvalkeal We discussed the naming of URI vs URL yesterday on the channel. @markpollack and @trisberg were advising. Since we already use URI for config server, and an URI can be an URL, I think it would be for the best for us to switch to URI.

The original issue however is not due naming, is to the fact that we were using a full path `/collector/metrics/streams` and we should let this be done at the client tier. We have hypermedia support and link traversal should be possible, and once we get discoverability in the server, we may not even need this property.

**Comment by markpollack** [comment]
Squashed and merged as 70e9b6f1b21a6b7fae88b9861561960b73f67e02

**Pull request closed without merging**